pages/sponsorship.js

```diff
@@ -0,0 +1,301 @@
+import Card, { CardHeader, CardTitle, CardContent, CardButtonContainer, CardDiv, TableDiv } from '../components/card'
+import styled from 'styled-components'
+import Button from '../components/button'
+import React, { useState, useRef, useEffect } from 'react'
+import { EDIT, NEW, VIEW, DELETE, COLOR, SPONSORSHIP} from '../constants'
+import fireDb, { db } from "../utility/firebase"
+import Modal, { Label, ModalContent, ModalField, LogoImage, UploadContainer } from '../components/modal'
+
+
+// TODO: move font family higher, reduce need to redefine
+
+const Text = styled.text`
+    padding-right: 80px;
+    flex: 1;
+    font-size: 16px;
+    white-space: nowrap;    
+    overflow: hidden;         
+    text-overflow: ellipsis;
+`
+const Actions = styled.div`
+    padding-right: 80px;
+    flex: 1;
+    display: flex;
+    justify-content: flex-start;
+`
+
+export default function SponsorshipPage(props) {
+
+    const inputFile = React.createRef();
+
+    const [sponsors, setSponsors] = useState({});
+
+    const [newobj, setnewobj] = useState({
+        name: "",
+        link: "",
+        imgURL: "", 
+        lastmod: "",
+        tier: ""
+    })
+    const [imgObject, setimgObject] = useState({});
+    const [imgFile, setimgFile] = useState({});
+    const [showEditWindow, setShowEditWindow] = useState(false);
+    const [modalAction, setmodalAction] = useState({});
```

Comment:
Nit: camelCase for consistency
```suggestion
    const [newobj, setNewObj] = useState({
        name: "",
        link: "",
        imgURL: "", 
        lastMod: "",
        tier: ""
    })
    const [imgObject, setImgObject] = useState({});
    const [imgFile, setImgFile] = useState({});
    const [showEditWindow, setShowEditWindow] = useState(false);
    const [modalAction, setModalAction] = useState({});
```